File: 프로그래머스/lv2/87390. n＾2 배열 자르기/n＾2 배열 자르기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, long long left, long long right)
{
    int len = right - left + 1;
    vector<int> answer;
    int r = left / n;
    int c = left % n;
    for (int i = 0; i < len; i++)
    {
        answer.push_back((r > c ? r : c) + 1);
        r += ++c / n;
        c %= n;
    }
    return answer;
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.java
class Solution {
    public int[] solution(int n, long left, long right) {
        int[] answer = new int[(int)(right - left + 1)];
        for (long idx = left; idx <= right; idx++) {
            long i = idx / n;
            long j = idx % n;
            answer[(int)(idx - left)] = (int)(Math.max(i, j) + 1);
        }
        return answer;
    }
}